core/gather/gatherers/root-causes.js

```diff
@@ -20,125 +19,25 @@ class RootCauses extends BaseGatherer {
-            response.nodeIds);
-        return nodeIds;
-      },
-      /** @param {import('@paulirish/trace_engine/generated/protocol.js').DOM.NodeId} nodeId */
-      async getNode(nodeId) {
-        try {
-          const response = await driver.defaultSession.sendCommand('DOM.describeNode', {nodeId});
-          // This always zero, so let's fix it here.
-          // https://bugs.chromium.org/p/chromium/issues/detail?id=1515175
-          response.node.nodeId = nodeId;
-          const node =
-            /** @type {import('@paulirish/trace_engine/generated/protocol.js').DOM.Node} */(
-              response.node);
-          return node;
-        } catch (err) {
-          if (err.message.includes('Could not find node with given id')) {
-            // TODO: when injecting an iframe, the engine gets the node of that frame's document element.
-            // But we don't have a way to access that frame. We just have our default session.
-            // Ex:
-            // node cli http://localhost:10503/shift-attribution.html --quiet --only-audits layout-shifts
-            // To fix we must:
-            // - Change trace engine `getNode` protocol interface to also give frame id
-            // - Expand our driver.targetManager to know how to talk to a session connected to a specific frame
-            // When this is fixed, remove this try/catch.
-            // Note: this could be buggy by giving the wrong node detail if a node id meant for a non-main frame
-            // happens to match one from the main frame ... which is pretty likely...
-            // TODO: fix trace engine type to allow returning null.
-            return /** @type {any} */(null);
-          }
-          throw err;
-        }
-      },
-      /** @param {number} nodeId */
-      async getComputedStyleForNode(nodeId) {
-        try {
-          const response = await driver.defaultSession.sendCommand(
-            'CSS.getComputedStyleForNode', {nodeId});
-          return response.computedStyle;
-        } catch {
-          return [];
-        }
-      },
-      /** @param {import('@paulirish/trace_engine/generated/protocol.js').DOM.NodeId} nodeId */
-      async getMatchedStylesForNode(nodeId) {
-        try {
-          const response = await driver.defaultSession.sendCommand(
-            'CSS.getMatchedStylesForNode', {nodeId});
-          return {...response, getError() {}};
-        } catch (err) {
-          return /** @type {any} */({getError() {
-            return err.toString();
-          }});
-        }
-      },
-      /** @param {string} url */
-      // @ts-expect-error not using, dont care about type error.
-      // eslint-disable-next-line no-unused-vars
-      async fontFaceForSource(url) {
-        return null;
-      },
-    };
+  async getArtifact(context) {
+    const trace = context.dependencies.Trace;
+    const traceEngineResult = await TraceEngineResult.request({trace}, context);
```

Comment:
Why even keep the gatherer at this point? It's just restructuring data from `TraceEngineResult` right?